addons/point_of_sale/models/pos_order.py

```diff
@@ -1066,6 +1066,7 @@ def _prepare_procurement_values(self, group_id=False):
             'partner_id': self.order_id.partner_id.id,
             'product_description_variants': self.full_product_name,
             'company_id': self.order_id.company_id,
+            'pos_order': True,
```

Comment:
I don't think you need this, can't you check later if in vals.get('group_id') pos_order_id is set or not?

Reply:
Is it correct if I check with `values.get('group_id')['pos_order_id']` or should I do it another way ?

Reply:
the thing is that you do a "get" to see if there is a group in the dictionary. If not, the 'get' function will return False, and False['pos_order_id'] will return a traceback.

So what you can do is values.get('group_id') and values.['group_id'].pos_order_id

as you'll know that there is a group, then you'll be able to take the values from the dictionary.

When you have doubt with small syntax tricks, don't hesitate to launch a python command in your terminal and test it with small dictionnay

Reply:
Thank you for the advice, the code has been fixed